### PR TITLE
i18n: stack-pack i18n

### PR DESCRIPTION
**Pull request by exterkamp**
**Summary**
Converts stack packs to use UIStrings for i18n.

### CODE REVIEW

**Comment by brendankenny** [comment]
> Summary

> Related Issues/PRs

can you fill this in, please :)

**Comment by brendankenny** [comment]
also, can we merge this into master after #7243 lands rather than continuing to grow that PR?

**Comment by exterkamp** [comment]
Sure, just land that one and I'll change this to merge 2 master.

File: stack-packs/packs/wordpress.js [hunk 1]
```diff
@@ -7,40 +7,62 @@
...
 
+const UIStrings = {
+  /** Additional description of a Lighthouse audit that tells the user *how* they can improve performance by removing unused CSS, in the context of the Wordpress CMS platform. This is displayed after a user expands the section to see more. No character length limits. Links in (parenthesis) become link texts to additional documentation. */
```
Comment (brendankenny) [inline stack-packs/packs/wordpress.js]: may not be a problem, but I'd say remove the emphasis on *how* in these strings, I imagine that could just end up confusing in the translation interface

File: stack-packs/packs/wordpress.js [hunk 2]
```diff
@@ -7,40 +7,62 @@
...
+  /** Additional description of a Lighthouse audit that tells the user *how* they can improve performance by minifying their Javascript files in the context of the Wordpress CMS platform. This is displayed after a user expands the section to see more. No character length limits. Links in (parenthesis) become link texts to additional documentation. */
+  unminified_javascript: 'A number of [WordPress plugins](https://wordpress.org/plugins/search/minify+javascript/) can speed up your site by concatenating, minifying, and compressing your scripts. You may also want to use a build process to do this minification up front if possible.',
+  /** Additional description of a Lighthouse audit that tells the user *how* they can improve performance by encoding animationg in HTML5, in the context of the Wordpress CMS platform. This is displayed after a user expands the section to see more. No character length limits. Links in (parenthesis) become link texts to additional documentation. */
```
Comment (brendankenny) [inline stack-packs/packs/wordpress.js]:
```suggestion
  /** Additional description of a Lighthouse audit that tells the user *how* they can improve performance by encoding animated images as video, in the context of the Wordpress CMS platform. This is displayed after a user expands the section to see more. No character length limits. Links in (parenthesis) become link texts to additional documentation. */
```

File: stack-packs/packs/wordpress.js (hunk 1 above)
Comment (exterkamp) [reply stack-packs/packs/wordpress.js]: Done.

File: stack-packs/packs/wordpress.js (hunk 2 above)
Comment (exterkamp) [reply stack-packs/packs/wordpress.js]: Done.

File: stack-packs/packs/wordpress.js [hunk 3]
```diff
@@ -7,40 +7,62 @@
 
 'use strict';
 
+const i18n = require('../../lighthouse-core/lib/i18n/i18n.js');
```
Comment (brendankenny) [inline stack-packs/packs/wordpress.js]: to fix Travis, I think we won't be able to call out from `stack-packs` code to `lighthouse`, at least as long as we're treating `stack-packs` as a separate module.

I think we'll have to remove `@lighthouse/stack-packs` from `package.json` and in `stack-packs.js` switch `const stackPacks = require('@lighthouse/stack-packs');` to `const stackPacks = require('../../stack-packs/index.js');`.

Maybe others have a different solution? Even as a separate module I'm not sure how we're going to have it call back into lighthouse when it's a lighthouse dep.

Comment (patrickhulce) [reply stack-packs/packs/wordpress.js]: > Maybe others have a different solution? Even as a separate module I'm not sure how we're going to have it call back into lighthouse when it's a lighthouse dep.

What if we turned this into a function and LH passed in the `i18n` object?

We could still keep UIStrings at the top level so we can statically extract them but the runtime piece is addressed at least.

**Comment by exterkamp** [comment]
I'm extracting these and sending for translation!

**Comment by exterkamp** [comment]
Changed based on @brendankenny's ideas.  I didn't commit the yarn.lock changes b/c i couldn't remember if that was just my local getting messed up with all this git merging or not.

File: stack-packs/packs/wordpress.js (hunk 3 above)
Comment (paulirish) [reply stack-packs/packs/wordpress.js]: > What if we turned this into a function and LH passed in the i18n object?

interesting idea. i just tried it and it does work.

```diff
diff --git a/lighthouse-core/lib/stack-packs.js b/lighthouse-core/lib/stack-packs.js
index b2a04fc4..591eb91b 100644
--- a/lighthouse-core/lib/stack-packs.js
+++ b/lighthouse-core/lib/stack-packs.js
@@ -6,8 +6,8 @@
 'use strict';
 
 const stackPacks = require('../../stack-packs/index.js');
+const i18n = require('./i18n/i18n.js');
 const log = require('lighthouse-logger');
-
 /**
  * Pairs consisting of a stack pack's ID and the set of stacks needed to be
  * detected in a page to display that pack's advice.
@@ -35,12 +35,13 @@ function getStackPacks(pageStacks) {
     }
 
     // Grab the full pack definition
-    const matchedPack = stackPacks.find(pack => pack.id === stackPackToIncl.packId);
-    if (!matchedPack) {
+    const matchedPackFn = stackPacks.find(pack => pack.id === stackPackToIncl.packId);
+    if (!matchedPackFn) {
       log.warn('StackPacks',
         `'${stackPackToIncl.packId}' stack pack was matched but is not found in stack-packs lib`);
       continue;
     }
+    const matchedPack = matchedPackFn(i18n);
 
     packs.push({
       id: matchedPack.id,
diff --git a/stack-packs/packs/wordpress.js b/stack-packs/packs/wordpress.js
index a15194ac..e5b204b6 100644
--- a/stack-packs/packs/wordpress.js
+++ b/stack-packs/packs/wordpress.js
@@ -7,8 +7,6 @@
 
 'use strict';
 
-const i18n = require('../../lighthouse-core/lib/i18n/i18n.js');
-
 
@@ -42,9 +40,11 @@ const UIStrings = {
   time_to_first_byte: 'Themes, plugins, and server specifications all contribute to server response time. Consider finding a more optimized theme, carefully selecting an optimization plugin, and/or upgrading your server.',
 };
 
-const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
 
-module.exports = {
+module.exports = function(i18n) {
+  const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+return {
   id: 'wordpress',
   iconDataURL: wordpressIcon,
   title: 'WordPress',
@@ -65,4 +65,5 @@ module.exports = {
     'time-to-first-byte': str_(UIStrings.time_to_first_byte),
   },
 };
+};
 module.exports.UIStrings = UIStrings;
```

**Comment by paulirish** [comment]
Looks good. @brendankenny ya?

**Pull request merged**